### ISSUE TITLE
As a user I want a simple way to discard uncommitted changes because I want to prototype something and be able to throw away the idea easily

### ISSUE THREAD

**Issue by togakangaroo** [togakangaroo/ezgit#1]
Now that in itself is already a big win as far as I’m concerned if it covers all the edge cases we talked about above and is a single command with no flags

^ MVP?

But there's more. So if the idea here is to use this for prototyping ideas…well sure, you might want to throw it away or keep it…but its also useful to have checkpoints *while* prototyping. So what if you could have a prototyping workflow

```
ezgit prototype  (start workflow. This mode won't use an index and auto-tracks everything)
ezgit checkpoint   (no message required, but optional)
ezgit checkpoint   (also you dont need to worry about adding stuff since it really is about making checkpoints)
ezgit undo      (drop all changes in the working directory, they were dumb)
ezgit checkpoint   (I liked that last batch)
ezgit prototype done 
```

The last command would launch a new workflow to resolve what you want to do with the prototype queue
Do you want to 
* kill it as if it never happened? 
* Move the entire queue to a branch for later investigation?
* Turn it into one or a series of commits?

When you’re done, you're back in whatever mode you were in before because you can stack prototyping workflows

**Comment by togakangaroo** [comment]
@KuriosCurious I supposed I'd be ok with that, though that's also achievable by just moving to that point in history and then doing this.

**Comment by KuriosCurious** [comment]
yes but to your point: `You start with “I want to do this…” and it leads you down the right path rather than having to say “I want to do this, lets sit down and research what the right feature to do that is and then try to make sure I’m using it correctly”`
By taking care of that for the user, you don't run the risk of accidentally screwing up things outside of the prototype as moving to that point in history could be done in several ways and some are more destructive than others.
